src/LocalInstaller.ts

```diff
@@ -11,17 +12,31 @@ interface PackageByDirectory {
     [directory: string]: PackageJson;
 }
 
+export interface Env {
+    [name: string]: string;
+}
+
+export interface Options {
+    npmEnv?: Env;
+}
+
 export interface ListByPackage {
     [key: string]: string[];
 }
 
 export class LocalInstaller extends EventEmitter {
 
     private sourcesByTarget: ListByPackage;
+    private options: Options;
 
-    constructor(sourcesByTarget: ListByPackage) {
+    constructor(sourcesByTarget: ListByPackage, options?: Options) {
         super();
         this.sourcesByTarget = resolve(sourcesByTarget);
+        if (options) {
```

Comment:
Could we use Object.assign here? `options = Object.assign({}, options);`

Reply:
sure, good point

Reply:
`Object.assign` can handle an `undefined` value. So you can even remove this entire `if`:

```ts
 constructor(sourcesByTarget: ListByPackage, options?: Options) {
         super();
         this.sourcesByTarget = resolve(sourcesByTarget);
         this.options = Object.assign({}, options);
     }
```